turbo/snapshotsync/freezeblocks/block_reader.go

```diff
@@ -359,7 +362,7 @@ func (r *BlockReader) HeaderByNumber(ctx context.Context, tx kv.Getter, blockHei
 	defer view.Close()
 	seg, ok := view.HeadersSegment(blockHeight)
 	if !ok {
-		return
+		return nil, errors.New("segment not found")
```

Comment:
we should revert this change - this function has many many usages that rely on the return value for a header to be nil and the error to be nil